Necroa's salvage update

Necro-a opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? --> Added a new Salvage facility: Geneticore, for the Shadow biome, as well as 2 new enemy types, Flesh and Elementals. Also added more drops to the expedition table. I also switched the Shadow biomes chasms for shadow anomaly shadows.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. --> I have always felt like salvage expeditions needed more variety and more exciting loot.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Negative, sir.
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.
https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/92a748d3-50b2-4e87-8a32-c34432a17703
Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/32e4cfbc-dd76-4afc-b028-b747e20c7d10)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/00fdee2a-89f4-4cdf-a6d1-a8856b453323)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/1ffc96b6-7888-449d-9713-aa6668d72747)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/bd811f62-d431-4228-b942-3c668ba4e42f)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/19d86b6e-fb4f-452d-a185-eb0bc7514811)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/f3c32b8f-a6be-4380-b00f-96e34dee5855)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/c3c357fc-e7f8-4297-9bd1-81588f0371d6)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/205290b0-351f-4acc-be98-675c026519c2)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/e5e9a402-a140-4ae2-9be4-1c9d8f11162b)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/1cf3f649-2920-46c2-92c6-a174d831d949)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/faa3d9d3-8cf4-4a06-80e5-657d0ba15ced)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/5b618752-c8a3-4bd1-8f35-fbe927e15b92)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/a967f976-3fe7-4646-8ed2-fa18d2cc3b56)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/e512a413-6910-46db-86b0-d6477af85722)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/50c2f106-6c76-463a-ae82-4ce0d2f506a5)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/578dcdc6-19b6-4cf8-9f02-8a20662ada71)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/99384149-45dc-47d3-aa2d-43ac9228821a)
![image](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/32c42808-cd1e-43f0-8333-746c9acf4096)
![Skärmbild 2024-05-17 162656](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/ae45256c-243c-4dca-bb19-1557953acaf6)
![Skärmbild 2024-05-17 162438](https://github.com/Sector-Umbra/Sector-Umbra/assets/163892215/46b4aebd-97fc-4ebc-a152-f017cdf2b251)

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
Added Geneticore facility
Reworked the shadow biome by replacing the chasms with shadow anomaly shadows
Added more expedition drops
Added Ore crabs and Flesh monsters to expedition enemy types.
<!--






-->

Simyon264 commented:
Could you please post screenshots of the additions?

Necro-a commented:
> Could you please post screenshots of the additions?

Is that good?

Necro-a commented:
Also, small note, this is the first time im doing something like this, keep that in mind

TsjipTsjip commented:
Something about this pull request is upsetting the linter. I'll have to poke around with the PR to see what exactly is going on.

Kukutis96513 commented:
how rare is this facility anyway? looking at the screenshots theres alot of loot in it.

Necro-a commented:
> Something about this pull request is upsetting the linter. I'll have to poke around with the PR to see what exactly is going on.

Fixed it, added the tag needed.

TsjipTsjip commented on Resources/Maps/Dungeon/geneticore.yml:
Should be in the _Umbra namespace.

TsjipTsjip commented on Resources/Prototypes/Entities/Mobs/NPCs/flesh.yml:
Should be added in a new file in the _Umbra namespace.

TsjipTsjip commented on Resources/Prototypes/Procedural/Themes/geneticore.yml:
Should be in the _Umbra namespace.

TsjipTsjip commented on Resources/Prototypes/Procedural/biome_templates.yml:
What do these changes do in-game? This looks like it affects things outside of the salvage dungeon being added.

Necro-a replied:
In game, all this does is replace the shadow biomes chasms with shadow kudzu shadows. i found it alot more fitting and less annoying to play. Granted this does affect the biome outside of expeditions as well but you can at least admeme using it without worrying about people getting round removed due to lag and stuff.

TsjipTsjip commented on Resources/Prototypes/Procedural/dungeon_configs.yml:
Should be added in a new file in the _Umbra namespace.

TsjipTsjip commented on Resources/Prototypes/Procedural/salvage_loot.yml:
Umbra comment implies only WeaponCrusher was added. Put Umbra comments on everything, or make it clear by stating `# START OF UMBRA ADDITIONS` and `# END OF UMBRA ADDITIONS` around the entire modified block.

This is gonna really really need a big balancing consideration as this also impacts other dungeons. I'm not sure I like this at first glance but we'll see how we go along.

TsjipTsjip commented on Resources/Prototypes/Procedural/salvage_mods.yml:
Separate file in _Umbra namespace, or Umbra comments.

TsjipTsjip commented on Resources/Prototypes/tags.yml:
No clue what this does, not my expertise, needs an _Umbra namespaced file tho.

TsjipTsjip commented on Resources/Prototypes/Procedural/salvage_factions.yml:
_Umbra namespace please.

TsjipTsjip reviewed (changes requested):
Lots of the same stuff. Have not tested this in-game yet. (How-to-test steps would be appreciated.)

Necro-a commented:
> Lots of the same stuff. Have not tested this in-game yet. (How-to-test steps would be appreciated.)

I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.

TsjipTsjip commented:
> I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.

I am not a salvager. I have not played salvage since months ago. Tell me how to test this biome in-game, what do I need to do in-game to test it.

Necro-a commented:
> > I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.
> 
> I am not a salvager. I have not played salvage since months ago. Tell me how to test this biome in-game, what do I need to do in-game to test it.

As in, doing an actual ingame expedition?

Necro-a commented:
> > I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.
> 
> I am not a salvager. I have not played salvage since months ago. Tell me how to test this biome in-game, what do I need to do in-game to test it.

1) Download the following: 
/Maps/_Umbra/Dungeon/geneticore.yml 
/Prototypes/_Umbra/Procedural/Themes/geneticore.yml
/Prototypes/_Umbra/Procedural/dungeon_configs.yml
/Prototypes/_Umbra/Procedural/salvage_mods.yml
/Prototypes/_Umbra/tags.yml

2) Put the files in the appropriate areas. (Aka just copy the file paths for the folders

3) Edit biome_templates.yml the same way its changed in the commits

4) Load the game up, spawn an expedition console, pick a shadow biome expedition.

5) Give yourself whatever gear you deem appropriate for a salvager to have, spawn a reclaimer (or any shuttle that can FTL.) and FTL to the expedition and play through it

Hope that helps.

TsjipTsjip commented:
Okay I didn't mean *that* in depth. What I wanted to know is if there's a way to force an expedition to generate with these settings, rather than having to get lucky on the expeditions console. (Far as I understand it, even shadow biome could be the regular shadow biome and thus not your facility.) I'll see if I can review this throughout next week. Timeline for merge is probably gonna be next sunday if no more major issues are found.

TsjipTsjip commented on Resources/Prototypes/Procedural/salvage_mods.yml:
```diff
@@ -248,3 +248,5 @@
   biomes:
     - Caves
+
+
```
This file should be left unchanged.

TsjipTsjip commented on Resources/Prototypes/Procedural/biome_templates.yml:
```diff
@@ -504,7 +504,7 @@
       allowedTiles:
       - FloorChromite
-      threshold: 0.2
+      threshold: 0.5
```
Add a `# Umbra` comment.

TsjipTsjip commented on Resources/Prototypes/Procedural/biome_templates.yml:
```diff
@@ -513,7 +513,7 @@
         gain: 0.4
       entities:
-      - FloorChromiteChasm
+      - ShadowKudzuWeak
```
Add a `# Umbra` comment.

TsjipTsjip reviewed (changes requested):
More minor remarks, this is already ready for full review tho. I'm gonna see if I can get that done throughout the following week.

Necro-a commented:
> Okay I didn't mean _that_ in depth. What I wanted to know is if there's a way to force an expedition to generate with these settings, rather than having to get lucky on the expeditions console. (Far as I understand it, even shadow biome could be the regular shadow biome and thus not your facility.) I'll see if I can review this throughout next week. Timeline for merge is probably gonna be next sunday if no more major issues are found.

You can forcefully generate a _dungeon_ but an entire expedition im not sure. Also, no worries, it should always generate a facility no matter what. If you wanna make sure you don't have to be lucky with the expedition console you can try removing the biome mods and dungeon mods from salvage_mods.yml so it ONLY shows the shadow biome.

SlamBamActionman commented:
Tested ingame and it seems to run just fine. 👍👍

There's probably something to be said about how the Elemental category is a pretty weak enemy type (especially since Salvage is the one job fully equipped to deal with ore crabs), but other than that it seems coolio.

Necro-a commented:
> Tested ingame and it seems to run just fine. 👍👍
> 
> There's probably something to be said about how the Elemental category is a pretty weak enemy type (especially since Salvage is the one job fully equipped to deal with ore crabs), but other than that it seems coolio.

Understood. Should i maybe add a salvage version of the ore crabs that are a bit stronger? Or maybe add specific types of reagent slimes to it so it matches in difficulty with the other enemy types?

TsjipTsjip commented:
Will have a proper look at this soon.

luckyshotpictures reviewed (approved):
Requested changes have been provided. Good to merge.

TsjipTsjip reviewed (approved):
Works in-game, drops were fixed. Sure, let's do this.
